pkg/assets/create/creater.go

```diff
@@ -0,0 +1,232 @@
+func fetchLatestDiscoveryInfo(dc *discovery.DiscoveryClient) (meta.RESTMapper, error) {
+	gr, err := restmapper.GetAPIGroupResources(dc)
+	if err != nil {
+		return nil, err
+	}
+	return restmapper.NewDiscoveryRESTMapper(gr), nil
+}
+
+func create(manifests map[string]*unstructured.Unstructured, client dynamic.Interface, mapper meta.RESTMapper, options CreateOptions) (error, bool) {
+	// Sort all manifests, so in case they use number prefixes, we follow their order, which will increase
+	// the chances for the create loop to succeed faster.
+	sortedManifestPaths := []string{}
+	for key := range manifests {
+		sortedManifestPaths = append(sortedManifestPaths, key)
+	}
+	sort.Strings(sortedManifestPaths)
+
+	// Record all errors for the given manifest path (so when we report errors, users can see what manifest failed).
+	errs := map[string]error{}
+
+	// In case we fail to find a rest-mapping for the resource, force to fetch the updated discovery on next run.
+	reloadDiscovery := false
+
+	for _, path := range sortedManifestPaths {
+		gvk := manifests[path].GetObjectKind().GroupVersionKind()
+		mappings, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			errs[path] = fmt.Errorf("unable to get REST mapping: %v", err)
+			reloadDiscovery = true
+			continue
+		}
+
+		if options.Verbose {
+			fmt.Fprintf(options.StdErr, "Creating %s ...\n", mappings.Resource.String())
+		}
+		if mappings.Scope.Name() == meta.RESTScopeNameRoot {
+			_, err = client.Resource(mappings.Resource).Create(manifests[path], metav1.CreateOptions{})
+		} else {
+			_, err = client.Resource(mappings.Resource).Namespace(manifests[path].GetNamespace()).Create(manifests[path], metav1.CreateOptions{})
+		}
+
+		// Resource already exists means we already succeeded
+		// This should never happen as we remove already created items from the manifest list, unless the resource existed beforehand.
+		if kerrors.IsAlreadyExists(err) {
+			delete(manifests, path)
+			continue
+		}
+
+		if err != nil {
+			errs[path] = fmt.Errorf("failed to create: %v", err)
+			continue
+		}
+
+		// Creation succeeded lets remove the manifest from the list to avoid creating it second time
+		delete(manifests, path)
+	}
+
+	return formatErrors("some manifests failed to create", errs), reloadDiscovery
+}
+
+func formatErrors(prefix string, errors map[string]error) error {
+	if len(errors) == 0 {
+		return nil
+	}
+	aggregatedErrMessages := []string{}
+	keys := []string{}
+	for key := range errors {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		aggregatedErrMessages = append(aggregatedErrMessages, fmt.Sprintf("%q: %v", k, errors[k]))
+	}
+	return fmt.Errorf("%s:\n%s", prefix, strings.Join(aggregatedErrMessages, "\n"))
+}
+
+func load(assetsDir string, options CreateOptions) (map[string]*unstructured.Unstructured, error) {
+	if _, err := os.Stat(assetsDir); os.IsNotExist(err) {
+		return nil, fmt.Errorf("directory %q does not exists", assetsDir)
```

Comment:
what about different errors?

Reply:
wanted to fail fast when the dir does not exists (i think file walk will probably explode with its own error?)

Reply:
iow. i think i can remove this